### PR TITLE
feat: oidc debugging for subscriptions

### PR DESCRIPTION
**Pull request by n1ru4l**
### Background

- We want to use GraphQL subscriptions within Hive
- We want to make OIDC integration debugging easier

### Description

- This adds a basic GraphQL Subscription setup using the GraphQL over SSE protocol

### Checklist

- [ ] Input validation
- [ ] Output encoding
- [ ] Authentication management
- [ ] Session management
- [ ] Access control
- [ ] Cryptographic practices
- [ ] Error handling and logging
- [ ] Data protection
- [ ] Communication security
- [ ] System configuration
- [ ] Database security
- [ ] File management
- [ ] Memory management
- [ ] Testing

### CODE REVIEW

File: packages/services/api/src/context.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ export interface RegistryContext {
   requestId: string;
   user: any;
   headers: Record<string, string | string[] | undefined>;
-  abortSignal: AbortSignal;
+  request: Request;
```
Comment (kamilkisiela) [inline packages/services/api/src/context.ts:8]: We now have `req` and `request`. Do you think we should merge these two?

File: packages/services/server/src/graphql-handler.ts [hunk 2]
```diff
@@ -268,37 +268,19 @@ export const graphqlHandler = (options: GraphQLHandlerOptions): RouteHandlerMeth
   return async (req, reply) => {
     const requestIdHeader = req.headers['x-request-id'] ?? reqIdGenerate();
     const requestId = cleanRequestId(requestIdHeader);
-    let controller = abortControllerCache.get(req.socket);
```
Comment (kamilkisiela) [inline packages/services/server/src/graphql-handler.ts:271]: no abort?

File: packages/services/api/src/context.ts (hunk 1 above)
Comment (n1ru4l) [reply packages/services/api/src/context.ts:8]: They were in the context before, I only added type for `request: Request;`, so I can re-use it for the abort signal

Comment (n1ru4l) [reply packages/services/api/src/context.ts:8]: I think we should replace all usages of `req` with `request` where possible

**Review by kamilkisiela — APPROVED** [review]

File: packages/services/server/src/index.ts [hunk 3]
```diff
@@ -154,6 +158,19 @@ export async function main() {
...
+  const redis = createRedisClient('Redis', env.redis, server.log.child({ source: 'Redis' }));
+
+  const pubSub = createPubSub({
+    eventTarget: createRedisEventTarget({
+      publishClient: redis,
+      subscribeClient: createRedisClient(
+        'subscriber',
+        env.redis,
+        server.log.child({ source: 'RedisSubscribe' }),
+      ),
```
Comment (n1ru4l) [inline packages/services/server/src/index.ts:170]: Subscriptions need a dedicated client as it blocks the Redis client.

File: packages/services/server/src/graphql-handler.ts (hunk 2 above)
Comment (n1ru4l) [reply packages/services/server/src/graphql-handler.ts:271]: we dont need to add our own abortController, we  already have the abortcontroller from whatwg-node/server on the context request object.

Comment (n1ru4l) [reply packages/services/server/src/graphql-handler.ts:271]: I changed the code for the mutex to use that one instead

**Pull request merged**